aten/src/ATen/native/cpu/LinearAlgebraKernel.cpp

```diff
@@ -0,0 +1,52 @@
+#include <ATen/ATen.h>
+#include <ATen/native/LinearAlgebra.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+namespace at { namespace native { namespace {
+
+void addr_kernel(TensorIterator &iter,
+                 Scalar beta, Scalar alpha) {
+  if (iter.dtype() == ScalarType::Bool) {
+    using scalar_t = bool;
+    auto beta_val = beta.to<scalar_t>();
+    auto alpha_val = alpha.to<scalar_t>();
+    cpu_kernel(iter,
+     [=](scalar_t self_val,
+         scalar_t vec1_val,
+         scalar_t vec2_val) __ubsan_ignore_undefined__ -> scalar_t {
+        return beta_val * self_val + alpha_val * vec1_val * vec2_val;
```

Comment:
this does not ignore `self_val` if beta=0, thus nans and infs in `self` won't be ignored, same in cuda kernel

Reply:
Thanks so much for pointing this out. I have updated the `CPU` and `cuda` kernels, the corresponding CPP and Pytorch test codes have been updated as well.